PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -1204,165 +1468,332 @@ TH1D* GetWeightsHist(const char* variable)
     if (!alien) {
-      cout << __LINE__ << endl;
-      exit(1);
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
     }
-    weightsFile = TFile::Open(Form("alien://%s", filePath), "READ");
-  } else {
-    // e) Handle the local case:
+    TFile *weightsFile = TFile::Open(Form("alien://%s", filePath), "READ");
+    if (!weightsFile) {
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
+    }
+
+    TList *baseList = NULL;
+    weightsFile->GetObject(
+        "ccdb_object", baseList); // TBI 20231008 for simplicity, harwired name
+                                  // of base TList is "ccdb_object" also for
+                                  // AliEn case, see if I need to change this
+    if (!baseList) {
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
+    }
+
+    TList *listWithRuns =
+        reinterpret_cast<TList *>(GetObjectFromList(baseList, runNumber));
+    if (!listWithRuns) {
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
+    }
+
+    hist = reinterpret_cast<TH1D *>(GetObjectFromList(
+        listWithRuns, Form("%s_%s", variable, fTaskName.Data())));
+    if (!hist) {
+      LOGF(fatal, "in function \033[1;31m%s at line %d\033[0m",
+           __PRETTY_FUNCTION__, __LINE__);
+    }
+
+  } else if (bFileIsInCCDB) {
+
+    // e) Handle the CCDB case: Remember that here I do not access the file,
+    // instead directly object in that file.
+    //    My home dir in CCDB:
+    //    https://alice-ccdb.cern.ch/browse/Users/a/abilandz/ Inspired by:
+    //    1. Discussion at:
+    //    https://alice-talk.web.cern.ch/t/access-to-lhc-filling-scheme/1073/17
+    //    2. See also:
+    //    https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/efficiencyGlobal.cxx
+    //                 https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/efficiencyPerRun.cxx
+    //    3. O2 Analysis Tutorial 2.0:
+    //    https://indico.cern.ch/event/1267433/timetable/#20230417.detailed
+
+    auto cm = o2::ccdb::BasicCCDBManager::instance();
+    cm.setURL("http://alice-ccdb.cern.ch");
+    // TH1F* hist = cm.get<TH1F>("Users/a/abilandz/test-2"); // this is the
+    // standard working example for histograms
+    if (fVerbose) {
+      LOGF(info, "\033[1;32mAccessing in CCDB %s\033[0m",
+           TString(filePath).ReplaceAll("/alice-ccdb.cern.ch/", "").Data());
+    }
+    TList *baseList =
+        (TList *)(cm.get<TList>(TString(filePath)
+                                    .ReplaceAll("/alice-ccdb.cern.ch/", "")
+                                    .Data()))
+            ->Clone(); // TBI 20231005 circumventing temporarily problem with
+                       // the ownership this way, but I shouldn't be really
+                       // clonning here anything
```

Comment:
You should not clone the CCDB object. You don't need the ownership, right? The ownership should be kept by the CCDB manager

Reply:
But be aware that you should not delete it either

Reply:
Fixed. Now there is a global instance of CCDB manager, instead of local.